pkg/virt-handler/migration-proxy/migration-proxy.go

```diff
@@ -347,7 +347,9 @@ func (m *migrationProxy) createUnixListener() error {
 func (m *migrationProxy) StopListening() {
 
 	close(m.stopChan)
-	m.listener.Close()
+	if m.listener != nil {
```

Comment:
Oh, that looks dangerous. Is that called from different go-routines? I think this needs to be synchronized. Close should be called when the channel is closed. Probably needs a fix on master too.

Reply:
When this is called from virt-handler, it's being called only from `StartSourceListener`, `StopSourceListener`, `StartTargetListener` or `StopTargetListener`, each one is synchronized.

It's also being used in virt-launcher, but always in the same thread.

Not sure why I need this change though ...

Reply:
> each one is synchronized.

very good.

> Not sure why I need this change though ...

Can it fail early where this is never set? E.g. create the proxy, defer Stop*Listener, then an unrelated error occurs, we naver call Start*Listener?

Reply:
Yeah, virt-launcher has a defer StopListener.
StartListening can return an error before it managed to set m.listener - so we are good here, I think :)